File: Backend/HRMS/HRMS.Application/DTOs/Core/DepartmentDto.cs
namespace HRMS.Application.DTOs.Core
{
    public class DepartmentDto
    {
        public int DeptId { get; set; }
        public string? DeptNameAr { get; set; }
        public string? DeptNameEn { get; set; }
        public int? ParentDeptId { get; set; }
        public string? ParentDeptNameAr { get; set; }
        public int BranchId { get; set; }
        public string? CostCenterCode { get; set; }
        public int? ManagerId { get; set; }
        public byte IsActive { get; set; }
    }

    public class CreateDepartmentDto
    {
        public required string DeptNameAr { get; set; }
        public string? DeptNameEn { get; set; }
        public int? ParentDeptId { get; set; }
        public int BranchId { get; set; }
        public string? CostCenterCode { get; set; }
        public int? ManagerId { get; set; }
        public int? IsActive { get; set; }
    }

    public class UpdateDepartmentDto
    {
        public required string DeptNameAr { get; set; }
        public string? DeptNameEn { get; set; }
        public int? ParentDeptId { get; set; }
        public int BranchId { get; set; }
        public string? CostCenterCode { get; set; }
        public int? ManagerId { get; set; }
        public int? IsActive { get; set; }
    }
}

File: Frontend/HRMS_Frontend/src/app/features/setup/models/department.dto.ts
// Matches HRMS.Application.DTOs.Core.DepartmentDto
export interface DepartmentDto {
  deptId: number;
  deptNameAr: string | null;
  deptNameEn: string | null;
  parentDeptId: number | null;
  parentDeptNameAr?: string | null;
  branchId: number;
  costCenterCode: string | null;
  managerId: number | null;
  isActive: number; // byte in C# -> number (0 or 1)
}

// Matches HRMS.Application.Features.Core.Departments.Commands.CreateDepartment.CreateDepartmentCommand
export interface CreateDepartmentCommand {
  deptNameAr: string;
  deptNameEn?: string | null;
  parentDeptId?: number | null;
  costCenterCode?: string | null;
  managerId?: number | null;
}

// Matches HRMS.Application.Features.Core.Departments.Commands.UpdateDepartment.UpdateDepartmentCommand
export interface UpdateDepartmentCommand {
  deptId: number;
  deptNameAr: string;
  deptNameEn?: string | null;
  parentDeptId?: number | null;
  costCenterCode?: string | null;
  managerId?: number | null;
  isActive: number;
}
